File: src/main/scala/com/example/modid/ExampleMod.scala
package com.example.modid

import scala.compiletime.uninitialized

import com.example.modid.proxy.IProxy
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent

import net.minecraftforge.fml.common.{Mod, SidedProxy}
import org.apache.logging.log4j.{LogManager, Logger}

@Mod(modid = Tags.MOD_ID, name = Tags.MOD_NAME, version = Tags.MOD_VERSION, modLanguage = "scala")
object ExampleMod {

  val LOGGER: Logger = LogManager.getLogger(Tags.MOD_NAME)

  @SidedProxy(modId = Tags.MOD_ID, clientSide = Tags.CLIENT_PROXY, serverSide = Tags.SERVER_PROXY)
  var proxy: IProxy = uninitialized


  /**
   * <a href="https://cleanroommc.com/wiki/forge-mod-development/event#overview">
   * Take a look at how many FMLStateEvents you can listen to via the @Mod.EventHandler annotation here
   * </a>
   */
  @Mod.EventHandler
  def preInit(event: FMLPreInitializationEvent): Unit = {
    proxy.preInit(event)
  }
}

File: src/main/java/com/example/modid/ExampleMod.java
package com.example.modid;

import com.example.modid.proxy.IProxy;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.SidedProxy;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

@Mod(modid = Tags.MOD_ID, name = Tags.MOD_NAME, version = Tags.MOD_VERSION)
public class ExampleMod {

    public static final Logger LOGGER = LogManager.getLogger(Tags.MOD_NAME);

    @SidedProxy(modId = Tags.MOD_ID, clientSide = Tags.CLIENT_PROXY, serverSide = Tags.SERVER_PROXY)
    public static IProxy proxy;

    /**
     * <a href="https://cleanroommc.com/wiki/forge-mod-development/event#overview">
     *     Take a look at how many FMLStateEvents you can listen to via the @Mod.EventHandler annotation here
     * </a>
     */
    @Mod.EventHandler
    public void preInit(FMLPreInitializationEvent event) {
        proxy.preInit(event);
    }
}
